docs/rfcs/0002_templates_mvp.md

```diff
@@ -0,0 +1,339 @@
+description, and owner of the source Template with the supplied values
+in the cloned Template.
+
+```graphql
+type CloneTemplateInput {
+    templateVersion: ID!
+    name: String!
+    description: String!
+    owner: ID
+}
+```
+
+### Permissions
+
+-   Any organization or study admin may create or clone Templates/TemplateVersions
+
+## 🛠 Modifying Templates
+
+### UpdateTemplateMutation
+
+```graphql
+type UpdateTemplateInput {
+    name: String
+    description: String
+    owner: ID
+}
+```
+
+### UpdateTemplateVersionMutation
+
+```graphql
+type UpdateTemplateVersionInput {
+    description: String
+    fieldsSchema: JSONString
+    fieldsSchemaFile: Upload
+}
+```
+
+### Permissions
+
+-   Any organization or study administrator may update a Template or
+    TemplateVersion if their organization owns it
+
+## 🗑 Deleting Templates
+
+### DeleteTemplateMutation
+
+```graphql
+type DeleteTemplateInput {
+    id: ID!
+}
+```
+
+### DeleteTemplateVersionMutation
+
+```graphql
+type DeleteTemplateVersionInput {
+    id: ID!
+}
+```
+
+### Permissions
+
+### Permissions
+
+-   Any organization or study administrator may delete a Template or
+    TemplateVersion if their organization owns it
+
+## 📦 Downloading Templates
+
+### Download Study Templates
+
+Probably the easiest way to download templates since it bundles everything
+into an archive. You will get an archive containing:
+
+For each TemplateVersion in the study:
+
+-   TemplateVersion.submission_template_file: the TSV file that people will fill out and use for data submission to their study
+-   TemplateVersion.fields_schema: the template field definitions which serve as instructions on filling out the template
+
+    /download/study/<Study.kf_id>templates
+
+We should consider providing a multi-sheet Excel workbook with a sheet for each
+template's fields schema and a folder of TSVs with all of the submission
+templates.
+
+### Download Latest Template
+
+Download the latest version of template.
+
+```
+/download/template/<Template.id>
+```
+
+You would get an archive with two files:
+
+-   Template.latest_version.submission_template_file
+-   Template.latest_version.fields_schema
+
+### Download a Template Version
+
+Download a specific version of a template.
+
+```
+/download/template/<Template.id>/versions/<TemplateVersion.id>
+```
+
+You would get an archive with two files:
+
+-   TemplateVersion.submission_template_file
+-   TemplateVersion.fields_schema
+
+### Permissions
+
+-   Any user may download template files
+
+## Changes to Existing Functionality
+
+-   A `Study` will have a new field, `Study.template_versions`, a many to many
+    relationship to `TemplateVersion`.
+-   The `File.file_type` field will be replaced by a one to one foreign key
+    to a `TemplateVersion`. The `TemplateVersion` must be one of
+    templates in the Study's template versions.
+-   The study and file mutations should be updated accordingly as well
```

Comment:
Right now we take all the column names and see which data types are subsets of them. This way we can tell the user which types are valid selections for the file.

Reply:
I don't know if that's a good idea. Could be but we might need to revise the user flow. Right now it's difficult to tell why a file type isn't showing up in the valid types list. 

For example, I upload a file that has 50 columns in it and it matches with lots of file types/templates: participant details, participant diseases, participant outcomes, participant ..., and specimen collection manifest. But for some reason the valid file types doesn't include the biobank manifest. Why? Ah its because I'm missing one column (e.g. Aliquot ID).

It might be better to allow the user to select the template they think their file conforms to. And then tell them if their file is missing required columns.

Reply:
Could also let them know if there are other templates that their file _does_ conform to